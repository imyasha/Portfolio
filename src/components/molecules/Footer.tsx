import React from "react"
import styled from "styled-components"

const Foot = styled.footer`
  padding: 0 3%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const Footer = () => {
  return (
    <Foot>
      <div>
        &copy; 2019 Jacob Herper
        <br />
        Proudly built with{" "}
        <a href="https://gatsbyjs.org/" target="_blank">
          Gatsby
        </a>
        .
      </div>
      <div>Social Links ...</div>
    </Foot>
  )
}

export default Footer
