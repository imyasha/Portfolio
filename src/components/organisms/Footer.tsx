import React from 'react';
import styled from 'styled-components';
import Menu from 'Atoms/Menu';

const Foot = styled.footer`
  padding: 0 3% 1rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  @media print {
    display: none;
  }

  @media screen and (max-width: 768px) {
    flex-direction: column;
    text-align: center;
  }
`;

const Footer = () => (
  <Foot>
    <div>
      &copy; {new Date().getFullYear()} Jacob Herper
      <br />
      Proudly built with{' '}
      <a href="https://gatsbyjs.org/" target="_blank" rel="noopener noreferrer">
        Gatsby
      </a>
      .
    </div>
    <nav>
      <Menu data-testid="footer-menu">
        <li data-testid="github">
          <a
            href="https://github.com/jakeherp"
            target="_blank"
            rel="noopener noreferrer"
          >
            Github
          </a>
        </li>
        <li data-testid="linkedin">
          <a
            href="https://www.linkedin.com/in/jacobherper/"
            target="_blank"
            rel="noopener noreferrer"
          >
            LinkedIn
          </a>
        </li>
        <li data-testid="twitter">
          <a
            href="https://twitter.com/jakeherp"
            target="_blank"
            rel="noopener noreferrer"
          >
            Twitter
          </a>
        </li>
      </Menu>
    </nav>
  </Foot>
);

export default Footer;
