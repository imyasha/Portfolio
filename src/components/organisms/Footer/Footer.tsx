import { SocialIcons } from 'Molecules/SocialIcons';

const Footer = () => {
	return (
		<footer className="flex justify-between mt-8 md:mt-20 px-4 md:px-20 py-8">
			<span>&copy; {new Date().getFullYear()} Jacob Herper</span>
			<SocialIcons
				profiles={[
					{
						name: 'Github',
						url: 'https://www.github.com/jakeherp',
						icon: 'GITHUB',
					},
					{
						name: 'LinkedIn',
						url: 'https://www.linkedin.com/in/jacobherper',
						icon: 'LINKEDIN',
					},
					{
						name: 'Twitter',
						url: 'https://twitter.com/jakeherp',
						icon: 'TWITTER',
					},
					{
						name: 'Instagram',
						url: 'https://www.instagram.com/jakeherp',
						icon: 'INSTAGRAM',
					},
				]}
			/>
		</footer>
	);
};

export { Footer };
