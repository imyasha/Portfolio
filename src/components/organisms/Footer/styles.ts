import styled from 'styled-components';

const StyledFooter = styled.footer`
	padding: 2rem 0;
`;

const Flex = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column-reverse;

	@media screen and (min-width: 768px) {
		flex-direction: row;
		justify-content: space-between;
	}
`;

export { Flex, StyledFooter };
