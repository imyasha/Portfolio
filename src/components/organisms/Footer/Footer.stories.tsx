import { Footer } from './Footer';

export default {
	title: 'organisms/Footer',
	component: Footer,
};

export const FooterStory = () => <Footer />;

FooterStory.storyName = 'Footer';
