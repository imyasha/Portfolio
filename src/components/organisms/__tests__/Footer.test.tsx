import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import Footer from '../Footer';

test('renders correctly', () => {
  const { container } = render(<Footer />);
  expect(container.firstChild).toMatchInlineSnapshot(`
    .c1 {
      list-style: none;
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: row;
      -ms-flex-direction: row;
      flex-direction: row;
      font-size: 20px;
      margin: 0;
    }

    .c1 li {
      margin: 0 0.5rem;
    }

    .c1 li a {
      padding: 0px 0.25rem;
    }

    .c1 li a::after {
      content: '';
      display: block;
      height: 10px;
      width: 0%;
      margin-top: -10px;
      background: #e94e1b;
      -webkit-transition: all 0.5s ease 0s;
      transition: all 0.5s ease 0s;
    }

    .c1 li a:hover {
      border-bottom: none;
    }

    .c1 li a:hover::after {
      width: 110%;
    }

    .c1 li svg {
      margin-left: 1rem;
    }

    .c0 {
      padding: 0 3% 1rem;
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
      -webkit-flex-direction: row;
      -ms-flex-direction: row;
      flex-direction: row;
      -webkit-box-pack: justify;
      -webkit-justify-content: space-between;
      -ms-flex-pack: justify;
      justify-content: space-between;
    }

    @media print {
      .c0 {
        display: none;
      }
    }

    @media screen and (max-width:768px) {
      .c0 {
        -webkit-flex-direction: column;
        -ms-flex-direction: column;
        flex-direction: column;
        text-align: center;
      }
    }

    <footer
      class="c0"
    >
      <div>
        Â© 
        2020
         Jacob Herper
        <br />
        Proudly built with
         
        <a
          href="https://gatsbyjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Gatsby
        </a>
        .
      </div>
      <nav>
        <ul
          class="c1"
        >
          <li
            data-testid="github"
          >
            <a
              href="https://github.com/jakeherp"
              rel="noopener noreferrer"
              target="_blank"
            >
              Github
            </a>
          </li>
          <li
            data-testid="linkedin"
          >
            <a
              href="https://www.linkedin.com/in/jacobherper/"
              rel="noopener noreferrer"
              target="_blank"
            >
              LinkedIn
            </a>
          </li>
          <li
            data-testid="twitter"
          >
            <a
              href="https://twitter.com/jakeherp"
              rel="noopener noreferrer"
              target="_blank"
            >
              Twitter
            </a>
          </li>
        </ul>
      </nav>
    </footer>
  `);
});
