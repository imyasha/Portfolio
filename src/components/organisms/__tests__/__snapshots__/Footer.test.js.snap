// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="sc-AxirZ DJfuB"
>
  <div>
    Â© 2019 Jacob Herper
    <br />
    Proudly built with
     
    <a
      href="https://gatsbyjs.org/"
      target="_blank"
    >
      Gatsby
    </a>
    .
  </div>
  <nav>
    <ul
      className="sc-AxjAm qarpn"
    >
      <li>
        <a
          href="https://github.com/jakeherp"
          target="_blank"
        >
          Github
        </a>
      </li>
      <li>
        <a
          href="https://www.linkedin.com/in/jacobherper/"
          target="_blank"
        >
          LinkedIn
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/jakeherp"
          target="_blank"
        >
          Twitter
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;
